enterprise/tpa_pipeline.py

```diff
@@ -78,7 +98,10 @@ def active_provider_enforces_data_sharing(request, enforcement_location):
         enforcement_location (str): the point where to see data sharing consent state.
         argument can either be "optional", 'at_login' or 'at_enrollment'
     """
-    running_pipeline = request.session.get('partial_pipeline')
+    try:
+        running_pipeline = get_partial_pipeline(request)
+    except TypeError:
+        running_pipeline = request.session.get('partial_pipeline_token')
```

Comment:
What are the circumstances where `get_partial_pipeline()` does not work but `running_pipeline = request.session.get('partial_pipeline_token')` does? And aren't those different types? (`get_partial_pipeline` returns a `dict` but `request.session.get('partial_pipeline_token')` returns a `str`)

If this is for a test case, it should be mocked out or replaced with a fake (install a simple `third_party_auth` fake app in the test environment, full of `Mock`s), rather than being here in the regular code.